src/AggregateFunctions/AggregateFunctionIntervalLengthSum.cpp

```diff
@@ -32,35 +32,35 @@ namespace
 
         if (WhichDataType{args.begin()[0]}.idx != WhichDataType{args.begin()[1]}.idx)
             throw Exception(
-                "Illegal type " + args.begin()[0]->getName() + " and " + args.begin()[1]->getName() + " of arguments of aggregate function "
-                    + name + ", there two arguments should have same DataType",
+                "Illegal types " + args.begin()[0]->getName() + " and " + args.begin()[1]->getName() + " of arguments of aggregate function "
+                    + name + ", there two arguments should have same data types",
                 ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
 
         for (const auto & arg : args)
         {
             if (!isNativeNumber(arg) && !isDateOrDateTime(arg))
```

Comment:
`isDateOrDateTime` also checks `DateTime64`, use `isDate` and `isDateTime`. We are going to remove `isDateOrDateTime` at all in https://github.com/ClickHouse/ClickHouse/pull/24398 (see resolved discussion).